scrapy_poet/downloadermiddlewares.py

```diff
@@ -107,6 +137,16 @@ def process_response(
         and an injectable attribute,
         the user-defined ``cb_kwargs`` takes precedence.
         """
+        if self._skip_dependency_creation(request, response, spider):
```

Comment:
The implementation of `_skip_dependency_creation` does not seem to need the response object. I wonder if we can move this code to process_request, and get rid of the changes to is_callback_requiring_scrapy_response, i.e. emit the warning from a single place in code.

Reply:
> The implementation of _skip_dependency_creation does not seem to need the response object.

Good catch! It's curious as to why our flake8 hook hasn't captured this. Probably time to look at bumping out all the pre-commit hook versions (e.g. flake8 is already v6 but we're using v4).

Reply:
> I wonder if we can move this code to process_request, and get rid of the changes to is_callback_requiring_scrapy_response, i.e. emit the warning from a single place in code.

One concern of having it in `process_request()` is that some user's project with custom middlewares declared after `InjectionMiddleware` that replaces the `Request.callback` value could mess up the check.

It could rarely happen, but we still need to write about such peculiarity somewhere in the docs to prevent users from stumbling to it. I think having it in `process_response()` is worth the extra code diffs in this case.

Reply:
> One concern of having it in `process_request()` is that some user's project with custom middlewares declared after `InjectionMiddleware` that replaces the `Request.callback` value could mess up the check.

That’s a good point. And also explains why we need the warning in process request as well for the specific case of DummyResponse, since that does not generate a response at all.